File: exercise/week01/java/src/test/java/neural/SubmarineTests.java
package neural;

import org.junit.jupiter.api.Test;

import java.io.IOException;
import java.io.InputStream;
import java.nio.charset.StandardCharsets;
import java.util.List;

import static org.assertj.core.api.Assertions.assertThat;

class SubmarineTests {

    @Test
    void should_compute_example_from_mission() {
        List<String> lines = List.of(
                "forward 5",
                "down 5",
                "forward 8",
                "up 3",
                "down 8",
                "forward 2"
        );

        Submarine submarine = new Submarine();
        submarine.executeAll(lines);

        assertThat(submarine.getHorizontal()).isEqualTo(15);
        assertThat(submarine.getDepth()).isEqualTo(10);
        assertThat(submarine.finalPosition()).isEqualTo(150);
    }

    @Test
    void should_move_on_given_instructions() throws IOException {
        InputStream is = getClass().getResourceAsStream("/submarine.txt"); // Load the submarine.txt file from the resources folder
        String content = new String(is.readAllBytes(), StandardCharsets.UTF_8); // Read the content of the file as a string
        List<String> lines = content.lines().toList();

        Submarine submarine = new Submarine();
        submarine.executeAll(lines);

        int result = submarine.finalPosition();

        System.out.println("Horizontal: " + submarine.getHorizontal());
        System.out.println("Depth: " + submarine.getDepth());
        System.out.println("Access Coordinate: " + result);

        assertThat(result).isGreaterThan(0);
    }
}

File: exercise/week01/c#/Submarine.Tests/SubmarineTests.cs
using FluentAssertions;
using Xunit;
using Xunit.Abstractions;

namespace Neural.Tests
{
    public class SubmarineTests
    {
        private readonly ITestOutputHelper _output; //to be able to write to the console in xUnit, _ to indicate that this is a private field

        public SubmarineTests(ITestOutputHelper output)
        {
            _output = output; //initialize the output helper
        }

        [Fact] //fact is an attribute that marks a method as a test method in xUnit
        public void Should_Compute_Example_From_Mission()
        {
            var lines = new[]
            {
                "forward 5",
                "down 5",
                "forward 8",
                "up 3",
                "down 8",
                "forward 2"
            };

            var submarine = new Submarine();
            submarine.ExecuteAll(lines);

            submarine.Horizontal.Should().Be(15);
            submarine.Depth.Should().Be(10);
            submarine.FinalPosition().Should().Be(150);
        }

        [Fact]
        public void Should_Move_On_Given_Text_Instructions()
        {
            var lines = File.ReadAllLines("submarine.txt");

            var submarine = new Submarine();
            submarine.ExecuteAll(lines);

            var result = submarine.FinalPosition();

            _output.WriteLine($"Horizontal: {submarine.Horizontal}");
            _output.WriteLine($"Depth: {submarine.Depth}");
            _output.WriteLine($"Access Coordinate (horizontal x depth): {result}");

            result.Should().BeGreaterThan(0);
        }
    }
}